### PR TITLE
mm : modify supporting multi heap region

### PR DESCRIPTION
**Pull request by jeongchanKim**
implement up_addregion() for adding additional heap region which needs new region start and size
change config name for expansion : HEAP2_BASE -> HEAPx_BASE, HEAP2_SIZE -> HEAPx_SIZE

### CODE REVIEW

File: os/arch/arm/src/common/up_allocateheap.c [hunk 1]
```diff
@@ -200,3 +204,29 @@ void up_allocate_kheap(FAR void **heap_start, size_t *heap_size)
...
+ * Name: up_addregion
+ ****************************************************************************/
+#if CONFIG_MM_REGIONS > 1
+void up_addregion(void)
+{
+	char *mem_start = CONFIG_HEAPx_BASE;
+	char *mem_size = CONFIG_HEAPx_SIZE;
+
+	while(*mem_start && *mem_size) {
+		kumm_addregion((void *)strtol(mem_start, NULL, 16), (size_t)strtol(mem_size, NULL, 0));
```
Comment (sunghan-chang) [inline os/arch/arm/src/common/up_allocateheap.c]: 1. for mem_size, is 0 right for third arg?
2. Don't you need to split by ','?
3. Don't you need to set endptr??

File: os/arch/arm/src/common/up_allocateheap.c [hunk 2]
```diff
@@ -200,3 +204,29 @@ void up_allocate_kheap(FAR void **heap_start, size_t *heap_size)
...
+/****************************************************************************
+ * Name: up_addregion
+ ****************************************************************************/
+#if CONFIG_MM_REGIONS > 1
+void up_addregion(void)
+{
+	char *mem_start = CONFIG_HEAPx_BASE;
+	char *mem_size = CONFIG_HEAPx_SIZE;
+
+	while(*mem_start && *mem_size) {
```
Comment (sunghan-chang) [inline os/arch/arm/src/common/up_allocateheap.c]: There is no warning when those are not set properly.

**Comment by seinfra** [comment]
Code Rule Check Result:
os/arch/arm/src/common/up_allocateheap.c:217: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/arch/arm/src/common/up_allocateheap.c:219: ERROR: [SPC_M_SEP] space prohibited before that close parenthesis ')'
os/arch/arm/src/common/up_allocateheap.c:219: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

**Comment by seinfra** [comment]
Code Rule Check OK.

File: os/arch/arm/src/common/up_allocateheap.c [hunk 3]
```diff
@@ -200,3 +204,26 @@ void up_allocate_kheap(FAR void **heap_start, size_t *heap_size)
...
+/****************************************************************************
+ * Name: up_addregion
+ ****************************************************************************/
+#if CONFIG_MM_REGIONS > 1
+void up_addregion(void)
+{
+	char *mem_start = CONFIG_HEAPx_BASE;
+	char *mem_size = CONFIG_HEAPx_SIZE;
+
+	while (*mem_start && *mem_size) {
```
Comment (sunghan-chang) [inline os/arch/arm/src/common/up_allocateheap.c]: Instead of for loop, while is ok?
```
for (i=0; i < CONFIG_MM_REGION - 1; i++) {
```

**Comment by seinfra** [comment]
Code Rule Check OK.

**Pull request merged**